### PR TITLE
use correct formula for nix2 mode and normal-inverse gamma

### LINKED ISSUE #20: disagreement in `mode` of NormalInverseChisq and NormalInverseGamma
These are two equivalent parametrizations of the conjugate prior for a normal with unknown mean and variance.  Both are based on Murphy, and the mode for NIX2 was taken directly from Murphy, while the mode for NIG isn't given explicitly.  

```julia
julia> using Distributions

julia> using ConjugatePriors: NormalInverseChisq, NormalInverseGamma

julia> d = NormalInverseGamma(0., 1., 2, 2)
ConjugatePriors.NormalInverseGamma{Float64}(mu=0.0, v0=1.0, shape=2.0, scale=2.0)

julia> d2 = NormalInverseChisq(d)
ConjugatePriors.NormalInverseChisq{Float64}(μ=0.0, σ2=1.0, κ=1.0, ν=4.0)

julia> mode(d2)
(0.0, 1.3333333333333333)

julia> mode(d)
(0.0, 0.6666666666666666)
```

They _do_ agree if the formula that Murphy gives for the mode of the marginal posterior of the variance is used:

```julia
julia> margmode(d::NormalInverseChisq) = d.ν*d.σ2 / (d.ν+2)
margmode (generic function with 1 method)

julia> margmode(d2)
0.6666666666666666
```

Which suggests to me that the NIG formula is also using the marginal posterior.

### PR DESCRIPTION
**Pull request by kleinschmidt**
Fixes #20, using the formula from Murphy for the mode of the `NormalInverseChisq` distribution for that and for the equivalent `NormalInverseGamma`

### CODE REVIEW

**Comment by kleinschmidt** [comment]
@mschauer (somehow can't request review directly)

File: src/normalinversegamma.jl [hunk 1]
```diff
@@ -43,7 +43,7 @@ end
 
 function mode(d::NormalInverseGamma)
     mu = d.mu
-    sig2 = d.scale / (d.shape + 1.0)
+    sig2 = d.scale / (d.shape - 0.5)
```
Comment (mschauer) [inline src/normalinversegamma.jl]: This can became negative for small shape??

Comment (mschauer) [reply src/normalinversegamma.jl]: I don't think that can be correct by looking at `pdf(D, mode(D)... +- eps())`

Comment (kleinschmidt) [reply src/normalinversegamma.jl]: hm, good point. maybe the mode is at 0 in that case?

Comment (kleinschmidt) [reply src/normalinversegamma.jl]: ...or rather infinite...

Comment (kleinschmidt) [reply src/normalinversegamma.jl]: ...which makes no sense now that I've thought about it for more than 5 seconds

**Comment by kleinschmidt** [comment]
Okay I sat down and did the calculus, checked that it works, and updated the code and tests.  Here's the derivation: https://gist.github.com/kleinschmidt/0466d46aed1736b47a01e5b975e07185

**Comment by mschauer** [comment]
Diff without whitespaces: https://github.com/JuliaStats/ConjugatePriors.jl/pull/21/files?w=1

**Comment by mschauer** [comment]
Could we have a test also for `mode(d::NormalInverseGamma)`?

**Comment by kleinschmidt** [comment]
I've added that test, too. I don't know why the diff is showing all the tests changes too but they're already in there.  Probably has something to do with me messing up the rebase/force-push of the PR branch.

**Pull request merged**